src/persist-client/src/impl/machine.rs

```diff
@@ -984,32 +1076,61 @@ mod tests {
                         "compare-and-append" => {
                             let input = get_arg(&tc.args, "input").expect("missing input");
                             let batch = state.batches.get(input).expect("unknown batch");
-                            let mut write = client
-                                .open_writer::<String, (), u64, i64>(shard_id)
-                                .await
-                                .expect("invalid shard types");
-                            let (_, _) = write
-                                .machine
-                                .compare_and_append(batch, &write.writer_id)
-                                .await
-                                .expect("indeterminate")
-                                .expect("invalid usage")
-                                .expect("upper mismatch");
+                            let reuse_write_handle = get_arg(&tc.args, "reuse-write-handle")
```

Comment:
if we're going to have a long-lived one, I think just always use it and save the complexity of the arg